### PR TITLE
Add clientInfo triggered session property rules

### PR DESCRIPTION
**Pull request by aweisberg**
Just letting Travis run it right now

Test plan - Unit tests. Will deploy a custom build with this to make sure it works.

```
== RELEASE NOTES ==

General Changes
* Add support for specifying session properties via regex matching on client info using :doc:`/admin/session-property-managers`
```

depended by https://github.com/facebookexternal/presto-facebook/pull/1392

### CODE REVIEW

**Review by viczhang861 — COMMENTED** [review]
LGTM

File: presto-session-property-managers/src/main/java/com/facebook/presto/session/SessionMatchSpec.java [hunk 1]
```diff
@@ -35,6 +35,7 @@
     private final Optional<Pattern> sourceRegex;
     private final Set<String> clientTags;
     private final Optional<String> queryType;
+    private final Optional<Pattern> clientInfoRegex;
```
[the remark this answers is not in the document]
Comment (aweisberg) [reply presto-session-property-managers/src/main/java/com/facebook/presto/session/SessionMatchSpec.java:38]: Client info contains information supplied by other software that isn't controlled by the person deploying Presto. It isn't always possible/convenient to include usable information in username or source where these things should normally go.

File: presto-session-property-managers/src/main/java/com/facebook/presto/session/SessionMatchSpec.java [hunk 2]
```diff
@@ -79,6 +82,13 @@ public SessionMatchSpec(
             }
         }
 
+        if (clientInfoRegex.isPresent()) {
+            String clientInfo = context.getClientInfo().orElse("");
+            if (!clientInfoRegex.get().matcher(context.getClientInfo().get()).matches()) {
```
[the remark this answers is not in the document]
Comment (aweisberg) [reply presto-session-property-managers/src/main/java/com/facebook/presto/session/SessionMatchSpec.java]: Good catch! Thanks.

**Review by wenleix — APPROVED** [review]
LGTM.

File: presto-session-property-managers/src/main/java/com/facebook/presto/session/SessionMatchSpec.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (aweisberg) [reply presto-session-property-managers/src/main/java/com/facebook/presto/session/SessionMatchSpec.java]: Updated to use this.

**Review by viczhang861 — APPROVED** [review]

**Review by arhimondr — APPROVED** [review]

**Comment by wenleix** [comment]
merged #15635, thanks for the contribution!

**Pull request merged**